### PR TITLE
adds disableHockeyVersionCheck feature

### PR DESCRIPTION
**Pull request by pbodsk**
which allows you to bypass Hockey version validation

### CODE REVIEW

File: versions/1.1-beta/parse_project_settings.rb [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    # Check if build number from hockey is nil
+    if buildnumber.nil?
+      puts yellow "|- No build found on Hockey, proceeding with build"
+      next
+    end
+
+    hockey_version = buildnumber["version"]
+    hockey_build = buildnumber["build"].to_i
+
+    disable_hockey_version_check = project_settings['disableHockeyVersionCheck'] ? true : false
```
Comment (pbodsk) [inline versions/1.1-beta/parse_project_settings.rb]: This is where the actual change starts 😄

**Comment by pbodsk** [comment]
@nickskull, @chriscombs, @jakobmygind 

I've tried implementing the feature we talked about on slack today so you can bypass the hockey validation.

Granted, the check is a bit crude...just another `unless` outside the existing one but it seems to work. If you have better suggestions I'm more than willing to listen 😄

**Comment by nickskull** [comment]
I think the actual hockey check shouldn’t be performed at all?

**Comment by pbodsk** [comment]
@nickskull you mean completely remove it so it is _never_ executed?

**Comment by nickskull** [comment]
@pbodsk yeah, I mean, what’s the point of doing the check if you ignore the result in the end?

**Comment by pbodsk** [comment]
We don't ignore it always.

If `disable_hockey_version_check` is set to true, _then_ we ignore it.

But if `disable_hockey_version_check` is not set (the default), then we still do the check and fail if pre-conditions are invalid.

```
unless disable_hockey_version_check
       unless (Version.new(xcode_version) >= Version.new(hockey_version)) && (xcode_build > hockey_build)
         valid = false
         warning_message = """#{key}: Xcode version #{xcode_version} (#{xcode_build}) is lower or equal than the one on Hockey #{hockey_version} (#{hockey_build})."""
         message +="#{warning_message} \n"
         puts yellow "|- #{warning_message}"
       end
     end
```

**Comment by nickskull** [comment]
Of course :D I meant only when disabling the check

**Pull request merged**